File: src/main/java/com/chrisdempewolf/fields/pin/BoardPinFields.java
package com.chrisdempewolf.fields.pin;

import com.chrisdempewolf.fields.BaseFields;
import com.chrisdempewolf.fields.Fields;

public class BoardPinFields extends BaseFields {

    @Override
    public Fields setAll() {
        return this.setId().setName().setUrl();
    }

    public BoardPinFields setId() {
        fields.add("id");
		return this;
    }

    public BoardPinFields setName() {
        fields.add("name");
		return this;
    }

    public BoardPinFields setUrl() {
        fields.add("url");
		return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        BoardPinFields that = (BoardPinFields) o;

        return !(fields != null ? !fields.equals(that.fields) : that.fields != null);

    }

    @Override
    public int hashCode() {
        return fields != null ? fields.hashCode() : 0;
    }

}

File: src/main/java/com/chrisdempewolf/pinterest/fields/pin/BoardPinFields.kt
package com.chrisdempewolf.pinterest.fields.pin

import com.chrisdempewolf.pinterest.fields.BaseFields
import com.chrisdempewolf.pinterest.fields.Fields

class BoardPinFields : BaseFields() {

    override fun withAll(): Fields
            = this.withID().withName().withURL().withCounts()

    fun withID(): BoardPinFields {
        fields = fields.plus("id")
        return this
    }

    fun withName(): BoardPinFields {
        fields = fields.plus("name")
        return this
    }

    fun withURL(): BoardPinFields {
        fields = fields.plus("url")
        return this
    }

    fun withCounts(): BoardPinFields {
        fields = fields.plus("counts")
        return this
    }
}
